diff -up linux-5.7/drivers/net/dummy.c.40~ linux-5.7/drivers/net/dummy.c
--- linux-5.7/drivers/net/dummy.c.40~	2020-06-04 16:03:17.600886153 +0200
+++ linux-5.7/drivers/net/dummy.c	2020-06-04 16:07:28.861840919 +0200
@@ -43,7 +43,7 @@
 
 #define DRV_NAME	"dummy"
 
-static int numdummies = 1;
+static int numdummies = 0;
 
 /* fake multicast ability */
 static void set_multicast_list(struct net_device *dev)
diff -up linux-5.7/include/uapi/linux/if_bonding.h.40~ linux-5.7/include/uapi/linux/if_bonding.h
--- linux-5.7/include/uapi/linux/if_bonding.h.40~	2020-06-01 01:49:15.000000000 +0200
+++ linux-5.7/include/uapi/linux/if_bonding.h	2020-06-04 16:03:17.600886153 +0200
@@ -82,7 +82,7 @@
 #define BOND_STATE_ACTIVE       0   /* link is active */
 #define BOND_STATE_BACKUP       1   /* link is backup */
 
-#define BOND_DEFAULT_MAX_BONDS  1   /* Default maximum number of devices to support */
+#define BOND_DEFAULT_MAX_BONDS  0   /* Default maximum number of devices to support */
 
 #define BOND_DEFAULT_TX_QUEUES 16   /* Default number of tx queues per device */
 
