diff -up linux-5.7/arch/x86/include/asm/processor.h.41~ linux-5.7/arch/x86/include/asm/processor.h
diff -up linux-5.7/arch/x86/um/asm/processor.h.41~ linux-5.7/arch/x86/um/asm/processor.h
--- linux-5.7/arch/x86/um/asm/processor.h.41~	2020-06-04 16:08:38.359215863 +0200
+++ linux-5.7/arch/x86/um/asm/processor.h	2020-06-04 16:08:59.824339370 +0200
@@ -24,7 +24,7 @@
 /* REP NOP (PAUSE) is a good thing to insert into busy-wait loops. */
 static __always_inline void rep_nop(void)
 {
-	__asm__ __volatile__("rep;nop": : :"memory");
+	__asm__ __volatile__("lfence": : :"memory");
 }
 
 static __always_inline void cpu_relax(void)
