diff -up linux-5.5/fs/namespace.c.42~ linux-5.5/fs/namespace.c
--- linux-5.5/fs/namespace.c.42~	2020-02-29 18:09:45.825969811 +0100
+++ linux-5.5/fs/namespace.c	2020-02-29 18:10:19.558971298 +0100
@@ -2478,7 +2478,7 @@ static void mnt_warn_timestamp_expiry(st
 
 		time64_to_tm(sb->s_time_max, 0, &tm);
 
-		pr_warn("%s filesystem being %s at %s supports timestamps until %04ld (0x%llx)\n",
+		pr_debug("%s filesystem being %s at %s supports timestamps until %04ld (0x%llx)\n",
 			sb->s_type->name,
 			is_mounted(mnt) ? "remounted" : "mounted",
 			mntpath,
