diff -up linux-5.5/drivers/net/wireless/intel/iwlwifi/fw/dbg.c.43~ linux-5.5/drivers/net/wireless/intel/iwlwifi/fw/dbg.c
--- linux-5.5/drivers/net/wireless/intel/iwlwifi/fw/dbg.c.43~	2020-02-29 18:10:39.642972184 +0100
+++ linux-5.5/drivers/net/wireless/intel/iwlwifi/fw/dbg.c	2020-02-29 18:11:18.674973905 +0100
@@ -2445,7 +2445,7 @@ int iwl_fw_start_dbg_conf(struct iwl_fw_
 		return -EINVAL;
 
 	if (fwrt->dump.conf != FW_DBG_INVALID)
-		IWL_WARN(fwrt, "FW already configured (%d) - re-configuring\n",
+		IWL_DEBUG_FW_INFO(fwrt, "FW already configured (%d) - re-configuring\n",
 			 fwrt->dump.conf);
 
 	/* Send all HCMDs for configuring the FW debug */
diff -up linux-5.5/drivers/net/wireless/intel/iwlwifi/pcie/trans.c.43~ linux-5.5/drivers/net/wireless/intel/iwlwifi/pcie/trans.c
--- linux-5.5/drivers/net/wireless/intel/iwlwifi/pcie/trans.c.43~	2020-02-29 18:10:39.642972184 +0100
+++ linux-5.5/drivers/net/wireless/intel/iwlwifi/pcie/trans.c	2020-02-29 18:11:39.439974821 +0100
@@ -934,7 +934,7 @@ void iwl_pcie_apply_destination(struct i
 		return;
 	}
 
-	IWL_INFO(trans, "Applying debug destination %s\n",
+	IWL_DEBUG_INFO(trans, "Applying debug destination %s\n",
 		 get_fw_dbg_mode_string(dest->monitor_mode));
 
 	if (dest->monitor_mode == EXTERNAL_MODE)
