--- linux-5.3/drivers/nvme/host/core.c.34~	2019-11-02 00:57:36.204395161 +0100
+++ linux-5.3/drivers/nvme/host/core.c	2019-11-02 00:53:37.052790183 +0100
@@ -2354,15 +2354,6 @@ static const struct nvme_core_quirk_entr
 		.vid = 0x2646,
 		.fr = "E8FK11.T",
 		.quirks = NVME_QUIRK_SIMPLE_SUSPEND,
-	},
-	{
-		/* https://forum.openmandriva.org/t/nvme-ssd-m2-not-seen-by-omlx-4-0/2407
-		 * Unknown SSD ... Maybe ADATA/Hynix (a similar mn from ADTA
-		 * but vid seems to be Hynix)
-		*/
-		.vid = 0x1c5c,
-		.mn = "HFM128GDHTNG-8310B",
-		.quirks = NVME_QUIRK_NO_APST,
 	}
 };
 
